common/primitives/key.go

```diff
@@ -143,54 +148,62 @@ func (c *PublicKey) Copy() (*PublicKey, error) {
 	return h, nil
 }
 
-func (a PublicKey) Fixed() [ed25519.PublicKeySize]byte {
-	return a
+// Fixed returns itself
+func (k PublicKey) Fixed() [ed25519.PublicKeySize]byte {
+	return k
 }
 
-func (a *PublicKey) IsSameAs(b *PublicKey) bool {
+// IsSameAs returns true iff input 'b' is the same as 'k'
+func (k *PublicKey) IsSameAs(b *PublicKey) bool {
 	if b == nil {
 		return false
 	}
-	for i := range a {
-		if a[i] != b[i] {
+	for i := range k {
+		if k[i] != b[i] {
 			return false
 		}
 	}
 	return true
 }
 
-func (pk *PublicKey) MarshalText() (rval []byte, err error) {
+// MarshalText marshals the public key into a []byte
+func (k *PublicKey) MarshalText() (rval []byte, err error) {
 	defer func(pe *error) {
 		if *pe != nil {
 			fmt.Fprintf(os.Stderr, "PublicKey.MarshalText err:%v", *pe)
 		}
 	}(&err)
-	return []byte(pk.String()), nil
+	return []byte(k.String()), nil
 }
 
-func (pk *PublicKey) UnmarshalText(b []byte) error {
+// UnmarshalText decodes the input []byte public key into 'pk'
```

Comment:
Not true anymore, because you renamed pk => k :smile:

Reply:
Fixed!